Large accuracy drop on Custom_Image_Classification_EfficientNet-V2-S with 2 GPUs

**Describe the bug**

while running regression testing, I can observe the large accuracy drop on Custom_Image_Classification_EfficientNet-V2-S.
evaluation score = 0.11111111
```bash
FAILED tests/regression/classification/test_classification.py::TestRegressionMultiClassClassification::test_otx_train[Custom_Image_Classification_EfficientNet-V2-S] - AssertionError: Performance: (0.1111111111111111) < Criteria: (0.6948)
```

**Steps to Reproduce**

1. prepare test datasets
2. CI_DATA_ROOT=/mnt/hdd1/data/ci_datasets tox -e tests-all-py310 -- tests/regression/classification/test_classification.py::TestRegressionMultiClassClassification::test_otx_train[Custom_Image_Classification_EfficientNet-V2-S]

**Environment:**

- OS: ubuntu 20.04
- Framework version: using tox
- Python version: 3.10
- OpenVINO version: using tox
- CUDA/cuDNN version: 11.7.1
- GPU model and memory: RTX3090 / 24G

--- a/src/otx/algorithms/action/adapters/mmaction/task.py
+++ b/src/otx/algorithms/action/adapters/mmaction/task.py
@@ -31,7 +31,6 @@ from mmcv.runner import CheckpointLoader, load_checkpoint, wrap_fp16_model
 from mmcv.utils import Config, ConfigDict, ProgressBar, get_git_hash
 from torch import distributed as dist
 
-from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.action.adapters.mmaction import (
     Exporter,
 )
@@ -50,6 +49,7 @@ from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
     MPAConfig,
     update_or_add_custom_hook,
 )
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.configs.configuration_enums import BatchSizeAdaptType
 from otx.algorithms.common.utils import append_dist_rank_suffix
 from otx.algorithms.common.utils.data import get_dataset

--- a/src/otx/algorithms/action/adapters/mmaction/task.py
+++ b/src/otx/algorithms/action/adapters/mmaction/task.py
@@ -31,6 +31,7 @@ from mmcv.runner import CheckpointLoader, load_checkpoint, wrap_fp16_model
 from mmcv.utils import Config, ConfigDict, ProgressBar, get_git_hash
 from torch import distributed as dist
 
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.action.adapters.mmaction import (
     Exporter,
 )
@@ -310,7 +311,7 @@ class MMActionTask(OTXActionTask):
         model.CLASSES = target_classes
 
         if cfg.distributed:
-            torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+            convert_sync_batchnorm(model)
 
         validate = bool(cfg.data.get("val", None))
 

--- a/src/otx/algorithms/classification/adapters/mmcls/task.py
+++ b/src/otx/algorithms/classification/adapters/mmcls/task.py
@@ -29,7 +29,6 @@ from mmcls.utils import collect_env
 from mmcv.runner import wrap_fp16_model
 from mmcv.utils import Config, ConfigDict
 
-from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms import TRANSFORMER_BACKBONES
 from otx.algorithms.classification.adapters.mmcls.utils.exporter import (
     ClassificationExporter,
@@ -58,6 +57,7 @@ from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
     MPAConfig,
     update_or_add_custom_hook,
 )
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.configs.configuration_enums import BatchSizeAdaptType
 from otx.algorithms.common.configs.training_base import TrainType
 from otx.algorithms.common.tasks.nncf_task import NNCFBaseTask

--- a/src/otx/algorithms/classification/adapters/mmcls/task.py
+++ b/src/otx/algorithms/classification/adapters/mmcls/task.py
@@ -29,6 +29,7 @@ from mmcls.utils import collect_env
 from mmcv.runner import wrap_fp16_model
 from mmcv.utils import Config, ConfigDict
 
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms import TRANSFORMER_BACKBONES
 from otx.algorithms.classification.adapters.mmcls.utils.exporter import (
     ClassificationExporter,
@@ -384,7 +385,7 @@ class MMClassificationTask(OTXClassificationTask):
         model.train()
 
         if cfg.distributed:
-            torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+            convert_sync_batchnorm(model)
 
         validate = bool(cfg.data.get("val", None))
         if validate:

--- a/src/otx/algorithms/common/adapters/torch/utils/__init__.py
+++ b/src/otx/algorithms/common/adapters/torch/utils/__init__.py
@@ -4,6 +4,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from .bs_search_algo import BsSearchAlgo
-from .utils import model_from_timm, convert_sync_batchnorm
+from .utils import convert_sync_batchnorm, model_from_timm
 
 __all__ = ["BsSearchAlgo", "model_from_timm", "convert_sync_batchnorm"]

--- a/src/otx/algorithms/common/adapters/torch/utils/__init__.py
+++ b/src/otx/algorithms/common/adapters/torch/utils/__init__.py
@@ -4,5 +4,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from .bs_search_algo import BsSearchAlgo
+from .utils import model_from_timm, convert_sync_batchnorm
 
-__all__ = ["BsSearchAlgo"]
+__all__ = ["BsSearchAlgo", "model_from_timm", "convert_sync_batchnorm"]

--- a/src/otx/algorithms/common/adapters/torch/utils/utils.py
+++ b/src/otx/algorithms/common/adapters/torch/utils/utils.py
@@ -4,8 +4,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import torch
-from torch.nn import Module
 from timm.models.layers import convert_sync_batchnorm as timm_cvt_sycnbn
+from torch.nn import Module
 
 
 def model_from_timm(model: Module) -> bool:
@@ -17,7 +17,7 @@ def model_from_timm(model: Module) -> bool:
     Returns:
         bool : whether model comes from timm or not.
     """
-    if "timm" in model.__module__.split('.'):
+    if "timm" in model.__module__.split("."):
         return True
 
     is_fisrt = True
@@ -34,7 +34,7 @@ def model_from_timm(model: Module) -> bool:
 
 def convert_sync_batchnorm(model: Module):
     """Convert BatchNorm layers to SyncBatchNorm layers.
-    
+
     Args:
         model (Module): model containing batchnorm layers.
     """

--- a/src/otx/algorithms/common/adapters/torch/utils/utils.py
+++ b/src/otx/algorithms/common/adapters/torch/utils/utils.py
@@ -1,0 +1,44 @@
+"""Collections of util functions related to torch."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from torch.nn import Module
+from timm.models.layers import convert_sync_batchnorm as timm_cvt_sycnbn
+
+
+def model_from_timm(model: Module) -> bool:
+    """Check a model comes from timm module.
+
+    Args:
+        model (Module): model to check it comes from timm module.
+
+    Returns:
+        bool : whether model comes from timm or not.
+    """
+    if "timm" in model.__module__.split('.'):
+        return True
+
+    is_fisrt = True
+    for sub_module in model.modules():
+        if is_fisrt:  # First module is the module itself.
+            is_fisrt = False
+            continue
+
+        if model_from_timm(sub_module):
+            return True
+
+    return False
+
+
+def convert_sync_batchnorm(model: Module):
+    """Convert BatchNorm layers to SyncBatchNorm layers.
+    
+    Args:
+        model (Module): model containing batchnorm layers.
+    """
+    if model_from_timm(model):
+        timm_cvt_sycnbn(model)
+    else:
+        torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)

--- a/src/otx/algorithms/common/adapters/torch/utils/utils.py
+++ b/src/otx/algorithms/common/adapters/torch/utils/utils.py
@@ -4,8 +4,12 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import torch
-from timm.models.layers import convert_sync_batchnorm as timm_cvt_sycnbn
 from torch.nn import Module
+
+try:
+    from timm.models.layers import convert_sync_batchnorm as timm_cvt_sycnbn
+except ImportError:
+    timm_cvt_sycnbn = None
 
 
 def model_from_timm(model: Module) -> bool:
@@ -38,7 +42,7 @@ def convert_sync_batchnorm(model: Module):
     Args:
         model (Module): model containing batchnorm layers.
     """
-    if model_from_timm(model):
+    if timm_cvt_sycnbn is not None and model_from_timm(model):
         timm_cvt_sycnbn(model)
     else:
         torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)

--- a/src/otx/algorithms/detection/adapters/mmdet/task.py
+++ b/src/otx/algorithms/detection/adapters/mmdet/task.py
@@ -32,7 +32,6 @@ from mmdet.datasets import build_dataloader, build_dataset, replace_ImageToTenso
 from mmdet.models.detectors import DETR, TwoStageDetector
 from mmdet.utils import collect_env
 
-from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.adapters.mmcv.hooks import LossDynamicsTrackingHook
 from otx.algorithms.common.adapters.mmcv.hooks.recording_forward_hook import (
     ActivationMapHook,
@@ -50,6 +49,7 @@ from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
     MPAConfig,
     update_or_add_custom_hook,
 )
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.configs.configuration_enums import BatchSizeAdaptType
 from otx.algorithms.common.configs.training_base import TrainType
 from otx.algorithms.common.tasks.nncf_task import NNCFBaseTask

--- a/src/otx/algorithms/detection/adapters/mmdet/task.py
+++ b/src/otx/algorithms/detection/adapters/mmdet/task.py
@@ -32,6 +32,7 @@ from mmdet.datasets import build_dataloader, build_dataset, replace_ImageToTenso
 from mmdet.models.detectors import DETR, TwoStageDetector
 from mmdet.utils import collect_env
 
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.adapters.mmcv.hooks import LossDynamicsTrackingHook
 from otx.algorithms.common.adapters.mmcv.hooks.recording_forward_hook import (
     ActivationMapHook,
@@ -278,7 +279,7 @@ class MMDetectionTask(OTXDetectionTask):
         model.CLASSES = target_classes
 
         if cfg.distributed:
-            torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+            convert_sync_batchnorm(model)
 
         validate = bool(cfg.data.get("val", None))
 

--- a/src/otx/algorithms/segmentation/adapters/mmseg/task.py
+++ b/src/otx/algorithms/segmentation/adapters/mmseg/task.py
@@ -32,6 +32,7 @@ from mmseg.apis import train_segmentor
 from mmseg.datasets import build_dataloader, build_dataset
 from mmseg.utils import collect_env
 
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.adapters.mmcv.hooks.recording_forward_hook import (
     BaseRecordingForwardHook,
     FeatureVectorHook,
@@ -374,7 +375,7 @@ class MMSegmentationTask(OTXSegmentationTask):
         model.CLASSES = target_classes
 
         if cfg.distributed:
-            torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+            convert_sync_batchnorm(model)
 
         validate = bool(cfg.data.get("val", None))
 

--- a/src/otx/algorithms/segmentation/adapters/mmseg/task.py
+++ b/src/otx/algorithms/segmentation/adapters/mmseg/task.py
@@ -32,7 +32,6 @@ from mmseg.apis import train_segmentor
 from mmseg.datasets import build_dataloader, build_dataset
 from mmseg.utils import collect_env
 
-from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.adapters.mmcv.hooks.recording_forward_hook import (
     BaseRecordingForwardHook,
     FeatureVectorHook,
@@ -48,6 +47,7 @@ from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
     MPAConfig,
     update_or_add_custom_hook,
 )
+from otx.algorithms.common.adapters.torch.utils import convert_sync_batchnorm
 from otx.algorithms.common.configs.configuration_enums import BatchSizeAdaptType
 from otx.algorithms.common.configs.training_base import TrainType
 from otx.algorithms.common.tasks.nncf_task import NNCFBaseTask

--- a/tests/unit/algorithms/common/adapters/torch/utils/test_bs_search_algo.py
+++ b/tests/unit/algorithms/common/adapters/torch/utils/test_bs_search_algo.py
@@ -1,9 +1,11 @@
 import pytest
 
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
 from otx.algorithms.common.adapters.torch.utils import BsSearchAlgo
 from otx.algorithms.common.adapters.torch.utils import bs_search_algo
 
 
+@e2e_pytest_unit
 class TestBsSearchAlgo:
     @pytest.fixture(autouse=True)
     def setup_test(self, mocker):

--- a/tests/unit/algorithms/common/adapters/torch/utils/test_utils.py
+++ b/tests/unit/algorithms/common/adapters/torch/utils/test_utils.py
@@ -1,0 +1,48 @@
+"""Tests for util functions related to torch."""
+
+from unittest.mock import  MagicMock
+
+import pytest
+
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+from otx.algorithms.common.adapters.torch.utils import utils as target_module
+from otx.algorithms.common.adapters.torch.utils import model_from_timm, convert_sync_batchnorm
+
+
+class OrdinaryModule:
+    def __init__(self):
+        self.module_arr = [MagicMock() for _ in range(3)]
+
+    def modules(self):
+        return self.module_arr
+
+
+def get_module(is_timm: bool = False):
+    module = OrdinaryModule()
+    if is_timm:
+        timm_sub_module = MagicMock()
+        timm_sub_module.__module__ = "timm.fake"
+        module.module_arr.append(timm_sub_module)
+
+    return module
+
+
+@e2e_pytest_unit
+@pytest.mark.parametrize("is_timm", [True, False])
+def test_model_from_timm(is_timm):
+    assert model_from_timm(get_module(is_timm)) is is_timm
+
+
+@e2e_pytest_unit
+@pytest.mark.parametrize("is_timm", [True, False])
+def test_convert_sync_batchnorm(mocker, is_timm):
+    mock_timm_cvt_sycnbn = mocker.patch.object(target_module, "timm_cvt_sycnbn")
+    mock_torch = mocker.patch.object(target_module, "torch")
+    model = get_module(is_timm)
+
+    convert_sync_batchnorm(model)
+
+    if is_timm:
+        mock_timm_cvt_sycnbn.assert_called_once()
+    else:
+        mock_torch.nn.SyncBatchNorm.convert_sync_batchnorm.assert_called_once()

--- a/tests/unit/algorithms/common/adapters/torch/utils/test_utils.py
+++ b/tests/unit/algorithms/common/adapters/torch/utils/test_utils.py
@@ -1,6 +1,6 @@
 """Tests for util functions related to torch."""
 
-from unittest.mock import  MagicMock
+from unittest.mock import MagicMock
 
 import pytest
 

